docs/proposals/20220825-azuremanaged-cluster-exp-graduation.md

```diff
@@ -0,0 +1,80 @@
+
+With the recent adoption of "`AzureManagedCluster`" by the capz community for practical, real-world use, we want to identify the set of outstanding items that may prevent graduation from experimental, and address each one of them, so that future adoption can be unlocked, and users can confidently build resilient systems on top of a stable API.
+
+## Motivation
+
+### Goals
+- Agree upon a durable, post-experimental specification of the set of CRDs that implement Managed Kubernetes on Azure, e.g., `AzureManagedCluster`, `AzureManagedControlPlane`, `AzureManagedMachinePool`
+- Prioritize timeliness of a post-experimental definition so that users can confidently plan for any breaking changes that result from a graduated spec as soon as possible
+- Prioritize "architectural affinity" with other Cluster API providers implementing Managed Kubernetes
+
+### Non-Goals / Future Work
+- Add to, or remove features from, "`AzureManagedCluster`"
+- Standardize any opinions about how best to use AKS
+- Promote capz-specific opinions about Managed Kubernetes across the Cluster API provider ecosystem
+- Define operational support
+
+## Post-graduation Prerequisites
+
+Concrete prereq workstreams are [tracked here](https://github.com/orgs/kubernetes-sigs/projects/26/views/1)
+
+### 1. Land Managed Cluster in Cluster API Proposal
+
+See:
+
+- https://github.com/kubernetes-sigs/cluster-api/pull/6988
+
+The above proposal defines a set of recommendations for Cluster API providers implementing Managed Kubernetes solutions. Because this proposal is an opt-in collection of architectural recommendations, it is not required that "`AzureManagedCluster`" strictly agrees with everything therein. However, by contributing to the successful landing of that proposal in Cluster API, we can best ensure that any capz-specific opinions or learnings are reflected, for the benefit of the larger ecosystem, and for the maximum happiness of AKS customers in particular.
+
+### 2. Integrate Cluster API Proposal Recommendations
+
+Once the proposal is accepted and merged into the Cluster API project as an endorsed set of provider recommendations, capz can audit the existing "`AzureManagedCluster`" experimental implementation for areas of disagreement with said recommendations. For each discovery of disagreement, there should be a defensible reason to matriculate a divergent capz Managed Kubernetes implementation into a post-experimental definition of "`AzureManagedCluster`", and ideally some form of sign-off from other provider maintainers. Where community consensus cannot be reached, capz should strongly consider evolving the experimental "`AzureManagedCluster`" implementation to meet the Cluster API Managed Kubernetes recommendation as a pre-requisite to graduating from experimental.
```

Comment:
Regarding the possibility of a successor to the CAPI proposal, would we block CAPZ's non-experimental managed cluster implementation on that?

Reply:
No. In fact there is no mention of a "Cluster API-native" ManagedCluster definition in this doc. Once a working group starts making progress towards this (assuming this proposal doc hasn't yet merged), it would probably make sense to mention that effort, and to explicitly state that AzureManagedCluster graduation is *not* dependent upon that proposal.

Off the top of my head I'd estimate that a long-term, capi ManagedCluster solution would be a `v1` thing (graduation of AzureManagedCluster will be within the `v1beta1` API scope).